File: data/code_pairs/fortran/2224.f90
MODULE cdf_aux_mod
  TYPE :: the_distribution
    INTEGER :: parameters(2)
  END TYPE the_distribution
  TYPE (the_distribution), PARAMETER :: the_beta = the_distribution((/99,999/))
CONTAINS
  SUBROUTINE set_bound(arg_name, test)
    INTEGER, INTENT (IN) :: arg_name, test
    if (arg_name .ne. test) STOP 'Bound check failed.'
  END SUBROUTINE set_bound
END MODULE cdf_aux_mod

MODULE cdf_beta_mod
  USE cdf_aux_mod
CONTAINS
  SUBROUTINE cdf_beta(which, test)
    INTEGER :: which, test
    CALL set_bound(the_beta%parameters(which), test)
  END SUBROUTINE cdf_beta
END MODULE cdf_beta_mod

PROGRAM test_cdf
  USE cdf_beta_mod
  IMPLICIT NONE
  CALL cdf_beta(1, 99)
  CALL cdf_beta(2, 999)
END PROGRAM test_cdf

File: data/code_pairs/cpp/2224.cpp
#include <iostream>
#include <array>
#include <stdexcept>

// Define the_distribution type
struct the_distribution {
    std::array<int, 2> parameters;
};

// Define the_beta constant
const the_distribution the_beta = {{99, 999}};

// Function to set bound
void set_bound(int arg_name, int test) {
    if (arg_name != test) {
        throw std::runtime_error("Bound check failed.");
    }
}

// Function cdf_beta
void cdf_beta(int which, int test) {
    set_bound(the_beta.parameters[which - 1], test);  // Adjust for 0-based index in C++
}

int main() {
    try {
        cdf_beta(1, 99);
        std::cout << "Test case 1 passed." << std::endl;

        cdf_beta(2, 999);
        std::cout << "Test case 2 passed." << std::endl;

        // This test is expected to throw an exception
        cdf_beta(1, 100);
    } catch (const std::runtime_error& e) {
        std::cout << "Test case 3 passed (caught expected error)." << std::endl;
    }
    return 0;
}